content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1210 @@
+
+However this meant that for a given point in time, the hierarchy could be "out of sync" and incorrect.
+
+Our solution to this problem is to remove the deferred "hierarchy maintenance system" in favor of making hierarchy changes "transactional". Hierarchy changes are now done via transactional [`Commands`], and directly modifying the component fields individually is no longer possible. This ensures that for a given point in time, the hierarchy is "correct".
+
+For most Bevy developers this is a non-breaking change, as most hierarchies are already constructed using `with_children` commands: 
+
+```rust
+commands
+    .spawn_bundle(SpatialBundle::default())
+    .with_children(|parent| {
+        parent.spawn_bundle(SpriteBundle {
+            texture: player_texture,
+            ..default()
+        });
+        parent.spawn_bundle(SpriteBundle {
+            texture: hat_texture,
+            ..default()
+        });
+    });
+```
+
+However for logic that adds/removes child entities from parents at runtime, the following commands must be used:
+```rust
+// removes the given children from the parent
+commands.entity(some_parent).remove_children(&[child1, child2]);
+// pushes the given children to the "end" of the parent's Children list
+commands.entity(some_parent).push_children(&[child3, child4]);
+// inserts the given children into the parent's Children list at the given index 
+commands.entity(some_parent).insert_children(1, &[child5]);
+``` 
+
+We've also added [`HierarchyEvent`], which makes it possible for developers to track changes in the hierarchy.
+
+There are still a couple of small holes to close, but staying on the "happy path" is much easier now:
+* removing only one of the components is possible (although heavily discouraged)
+* adding default values of only one of the components manually is still possible (although heavily discouraged)
+
+We're discussing ways to resolve this class of problem, such as [Archetype Rules / Invariants](https://github.com/bevyengine/bevy/issues/1481).
+
+[`Parent`]: https://docs.rs/bevy/0.8.0/bevy/hierarchy/struct.Parent.html
+[`Children`]: https://docs.rs/bevy/0.8.0/bevy/hierarchy/struct.Children.html
+[`HierarchyEvent`]: https://docs.rs/bevy/0.8.0/bevy/hierarchy/struct.HierarchyEvent.html
+
+
+## Taffy migration: a refreshed UI layout library
+
+<div class="release-feature-authors">authors: @alice-i-cecile, @jkelleyrtp, @Weibye, @TimJentzsch, @colepoirier</div>
+
+Bevy's moved off the abandoned [`stretch`](https://crates.io/crates/stretch) UI layout crate and onto its new community-maintained hard fork: `taffy`. Together with the [Dioxus](https://dioxuslabs.com/) team, we've dramatically cleaned up the code base, solved a critical performance issue with deep UI trees and freshened up the docs.
```

Comment:
```suggestion
Bevy's moved off the abandoned [`stretch`](https://crates.io/crates/stretch) UI layout crate and onto its new community-maintained hard fork: [`taffy`](https://crates.io/crates/taffy). Together with the [Dioxus](https://dioxuslabs.com/) team, we've dramatically cleaned up the code base, solved a critical performance issue with deep UI trees and freshened up the docs.
```

Reply:
I see there was some debate above as to whether to link stretch. But imho, if stretch is linked, taffy definitely should be.